rollover.py

```diff
@@ -0,0 +1,300 @@
+#! /usr/bin/env python
+
+import argparse
+import itertools
+from operator import itemgetter
+import os
+import sys
+
+# local imports
+import ec2
+import elb
+import ecs
+import scaling
+import ssh
+import utils
+
+
+def sort_and_check_availability_zones(instance_descriptions, asg_instances):
+    """
+    sorts the instances into an order that tries not to cause an AZ imbalance
+    when removing instances. Also, prompts the user if there are issues.
+    @param instance_descriptions: dict of ec2_ids to descriptions
+    @param asg_instances: list of asg instance dicts
+    @return: sorted list of instances to remove
```

Comment:
nit: some confusion over the term "instance". ec2_instance vs asg_instance .. my suggestion would be to just use instance when referring to ec2 instances and to call `asg_instances` -> `asgs`

e.g. the return says "a sorted list of instances" (I'm assuming ec2 instances)